googledataprocauthenticator/google.py

```diff
@@ -0,0 +1,319 @@
+                "to use for Application Default Credentials. Run `gcloud auth list` to see "\
+                "your credentialed accounts.".format(parsed_attributes.account))
+                raise new_exc
+            if self.active_credentials == 'default-credentials' and self.default_credentials_configured:
+                self.credentials, self.project = google.auth.default(scopes=self.scopes)
+            #fix account_dict
+            elif self.active_credentials in account_dict:
+                self.credentials = get_credentials_for_account(self.active_credentials, self.scopes)
+            else:
+                self.credentials, self.project = None, None
+        else:
+            if self.default_credentials_configured:
+                self.credentials, self.project = google.auth.default(scopes=self.scopes)
+                self.active_credentials = 'default-credentials'
+            #fix this to be active_credentials is in credentialed accounts 
+            elif active_user_account is not None:
+                self.credentials = get_credentials_for_account(active_user_account, self.scopes)
+                self.active_credentials = active_user_account
+            else:
+                self.credentials, self.project = None, None
+        Authenticator.__init__(self, parsed_attributes)
+        self.widgets = self.get_widgets(WIDGET_WIDTH)
+
+    def get_widgets(self, widget_width):
+        ipywidget_factory = IpyWidgetFactory()
+
+        self.project_widget = ipywidget_factory.get_text(
+            description='Project:',
+            width=widget_width
+        )
+
+        self.cluster_name_widget = ipywidget_factory.get_text(
+            description='Cluster:',
+            width=widget_width
+        )
+
+        self.region_widget = ipywidget_factory.get_text(
+            description='Region:',
+            width=widget_width
+        )
+
+        self.google_credentials_widget = ipywidget_factory.get_dropdown(
+            options=list_accounts_pairs(self.credentialed_accounts, self.default_credentials_configured),
+            value=None,
+            description=u"Account:"
+        )
+
+        if self.active_credentials is not None: 
+            self.google_credentials_widget.value = self.active_credentials
+        else: 
+            self.google_credentials_widget.disabled = True
+
+        widgets = [self.project_widget, self.region_widget, self.cluster_name_widget, self.google_credentials_widget]
+        return widgets
+
+    def initialize_credentials_with_auth_account_selection(self, account):
+        """Initializes self.credentials with the accound selected from the auth dropdown widget"""
+        if (account != self.active_credentials):
+            if (account == 'default-credentials'):
+                self.credentials, self.project = google.auth.default(scopes=self.scopes)
+                #self.credentials.refresh(self.callable_request)
```

Comment:
Nit: remove this here and below if unnecessary now. Looks like you're calling refresh elsewhere